src/OMSimulatorLib/SystemSC.cpp

```diff
@@ -806,15 +833,27 @@ oms_status_enu_t oms::SystemSC::doStepCVODE()
       // set states
       status = fmus[i]->setContinuousStates(states[i]);
       if (oms_status_ok != status) return status;
-
-      // set time
-      fmistatus = fmi2_setTime(fmus[i]->getFMU(), time);
-      if (fmi2OK != fmistatus) logError_FMUCall("fmi2_setTime", fmus[i]);
     }
 
     if (flag == CV_ROOT_RETURN || time == tnext)
     {
-      //logInfo("event found!!! " + std::to_string(time));
+      logDebug("event found!!! " + std::to_string(time));
+
+      // set time
+      for (const auto& component : getComponents())
```

Comment:
Aren't you concerned about duplicating the same code block 4 times and then having to successfully maintain this/fix bugs in this section, in the future? As opposed to, say, having this snippet in a small function called in 4 places?

Reply:
I completely agree! 😊

Reply:
@bilderbuchi feel free to comment on #1386

Reply:
❤️ great!